.github/workflows/build-and-deploy.yml

```diff
@@ -0,0 +1,65 @@
+name: build and deploy
+
+on:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: write
+  pages: write
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    outputs:
+      artifact-path: ${{ steps.build.outputs.out_dir }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          targets: wasm32-unknown-unknown
+
+      - name: Install wasm-pack
+        run: curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh
+
+      - name: Build WASM (api → client/src/wasm)
+        id: build
+        working-directory: api
+        run: |
+          mkdir -p ../client/src/wasm
+          wasm-pack build \
+            --release \
+            --target web \
+            --out-dir ../client/src/wasm
+          echo "::set-output name=out_dir::$(pwd)/../client"
+
+      - name: Setup Node.js (for client build)
+        uses: actions/setup-node@v3
+        with:
+          node-version: '20'
+
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 7
+
+      - name: Build client with Vite
+        working-directory: client
+        run: |
+          pnpm install
+          pnpm run build
+
+      - name: Publish to GitHub Pages
+        uses: peaceiris/actions-gh-pages@v4
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Using secrets its not so safe. Consider the approach based on enviroments, as an example take a look here

https://github.com/libamtrack/libamtrack.github.io/blob/main/.github/workflows/deploy.yml